### DISCUSSION TITLE
Does pyright --createstub work with compiled modules? (.pyd files)

### DISCUSSION THREAD

**Discussion by neelkarma** [microsoft/pyright#1664] [Q&A] [answered]
Or do I have to use stubgen for that? Also, if I do have to use stubgen for that, how would I get the generated type stubs to work with Pyright?

**Comment by erictraut** [comment, marked as answer]
No, pyright's --createstub doesn't work with compiled modules. You'll need to use stubgen for that, but the amount of type information available in a compiled module is small, so the resulting stubs will be very low quality. You'll need to fill in a lot of information manually to make them work for type checking. In my experience, writing such stubs from scratch is a better approach.
